pkg/obfuscator/azure_resources.go

```diff
@@ -0,0 +1,239 @@
+package obfuscator
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/openshift/must-gather-clean/pkg/schema"
+	"k8s.io/utils/set"
+)
+
+const (
+	maximumSupportedObfuscationsAzure = 9999999999
+
+	// Azure resource path templates
+	azureSubscriptionTemplate    = "x-subscription-%010d-x"
+	azureResourceGroupTemplate   = "x-resourcegroup-%010d-x"
+	azureResourceNameTemplate    = "x-resource-%010d-x"
+	azureSubresourceNameTemplate = "x-subresource-%010d-x"
+
+	staticAzureSubscriptionReplacement    = "obfuscated-subscription"
+	staticAzureResourceGroupReplacement   = "obfuscated-resourcegroup"
+	staticAzureResourceNameReplacement    = "obfuscated-resource-name"
+	staticAzureSubresourceNameReplacement = "obfuscated-subresource-name"
+)
+
+var (
+	//     /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Network/virtualNetworks/myVNet/subnets/mySubnet
+	// Azure resource path pattern
+	azureSubscriptionPattern  = `/subscriptions/([^(/\s')]+)`
+	azureResourceGroupPattern = `/resource[Gg]roups/([^(/\s')]+)`
+	azureResourcePattern      = `/providers/([^/]+)/([^/]+)/([^(/\s')]+)`
+	azureSubresourcePattern   = azureResourcePattern + `/([^/]+)/([^(/\s')]+)`
+	azureNodePoolPattern      = `Microsoft.RedHatOpenShift/hcpOpenShiftClusters/nodePools/([^(/\s')]+)`
```

Comment:
We'll need to be case insensitive here.  Resource IDs are case insensitive, including the strings like `subscription`, `resourcegroups`, etc.  

Sometimes they come back different values from ARM APIs, but they should be generally consistent.

Reply:
If you have a b- and Enterprise Managed User (EMU) account: https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/common-api-details.md#case-insensitivity-for-requests

Reply:
Fixed and tests added.